mesonbuild/compilers/c.py

```diff
@@ -710,3 +710,66 @@ def get_include_args(self, path: str, is_system: bool) -> T.List[str]:
         if path == '':
             path = '.'
         return ['--include_path=' + path]
+
+class TCCCCompiler(CCompiler):
+    def __init__(self, exelist: T.List[str], version: str, for_machine: MachineChoice,
+                 is_cross: bool, info: 'MachineInfo',
+                 exe_wrapper: T.Optional['ExternalProgram'] = None,
+                 linker: T.Optional['DynamicLinker'] = None,
+                 full_version: T.Optional[str] = None):
+        self.id = 'tcc'
+        CCompiler.__init__(self, exelist, version, for_machine, is_cross,
+                           info, exe_wrapper, linker=linker, full_version=full_version)
+        self.can_compile_suffixes.add('s')
+        self.base_options = {
+            OptionKey(o) for o in ['b_ndebug', 'b_staticpic']}
+
+    def get_options(self) -> 'KeyedOptionDictType':
+        opts = CCompiler.get_options(self)
+        key = OptionKey('std', machine=self.for_machine, lang=self.language)
+        opts[key].choices = ['none', 'c99', 'c11']
+        return opts
+
+    def get_always_args(self) -> T.List[str]:
+        return []
+
+    def get_option_compile_args(self, options: 'KeyedOptionDictType') -> T.List[str]:
+        args = []
+        std = options[OptionKey('std', lang=self.language, machine=self.for_machine)]
+        if std.value != 'none':
+            args.append('-std=' + std.value)
+        return args
+
+    def get_optimization_args(self, optimization_level: str) -> T.List[str]:
+        return []
+
+    def get_no_optimization_args(self) -> T.List[str]:
+        return []
+
+    def get_pic_args(self) -> T.List[str]:
+        return []
```

Comment:
It seems that -fPIC is not supported. Maybe it's worth throwing a MesonException here?

Reply:
Raising an exception here makes *a lot* of unit tests fail. I think even tests that compile shared libraries fail if i put an exception.